### ISSUE TITLE
Unused region checking

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#8319]
See https://github.com/flix/flix/blob/master/main/src/library/Environment.flix#L25-L31

regions are sometimes just used in annotations, mosty with interop where you add a region to something. This doesn't count as a variable usage so you have to write `let _ = rc;` to circumvent it.

**Comment by magnus-madsen** [comment]
In that example, I would argue the region should not even be there?

**Comment by JonathanStarup** [comment]
yes, but there could still be interop scenarios where you can't rely on `Array[_, IO]`
